### ISSUE TITLE
Match function ignores base path

### ISSUE THREAD

**Issue by valery-barysok** [dannyvankooten/AltoRouter#263]
Match function ignores base path value and use length instead of it:

BasePath: /123/
Matches to: /ANY/, /321/, /ETC/

**Comment by valery-barysok** [comment]
Possible fix is:

```
        $len = strlen($this->basePath)
        if (substr($requestUrl, 0, $len) !== $this->basePath) {
            return false;
        }

        // strip base path from request url
        $requestUrl = substr($requestUrl, strlen($this->basePath));
        $requestUrl = substr($requestUrl, $len);
```

**Comment by ThijmenDF** [comment]
Correct, as seen here. I assume it was done because of performance reasons.

https://github.com/dannyvankooten/AltoRouter/blob/f6fede4f94ced7c22ba63a9b8af0bf2dc38e3cb2/AltoRouter.php#L200

It would make more sense for it to compare the leading section, where it skips the route matching if the leading (or 'base path') part of the URL doesn't match the configured value.

What would be the desired effect? For it to simply match no routes if the base path doesn't match the URL?

**Comment by valery-barysok** [comment]
Sure. It must have predictable behaviour: an exact match with the base path. Otherwise, it introduces a security issue if the API is accidentally opened.

**Comment by JB-oclock** [comment]
is this #279 resolve it ?

**Comment by valery-barysok** [comment]
I think match function has to return false instead of an exception. Exception can be converted to a warning message.
